### DISCUSSION TITLE
updating a field by incrementing its value

### DISCUSSION THREAD

**Discussion by xu4wang** [directus/directus#17342] [Archive]
is it possible to update a field by incrementing its value? like the SQL below:

```sql
update TBL set CLM = CLM + 1 where key = 'KEY' 
```

It is supported by knex via knex.raw:

```javascript
knex('table_name')
  .where('key', 'KEY')
  .update('column_name', knex.raw('column_name + 1'))
  .then(() => {
    console.log('Record updated successfully!');
  })
  .catch((error) => {
    console.error('Error updating record: ', error);
  });
```

How can I do the same in Directus API? 

PS. 
- The background of the question is to have the read and write query in one SQL statement, so it will not be interrupted in a concurrent environment. 
- I understand I can create a partial item object and then call the [api](https://docs.directus.io/reference/items.html#update-an-item)

**Comment by br41nslug** [comment]
This is not something built into the API. You could give the feature request an upvote (https://github.com/directus/directus/discussions/3147) but looks like that never got much community interest. I have built an extension to do this you could give a try in the meantime  https://github.com/br41nslug/directus-extension-increment

**Reply by xu4wang** [reply]
> This is not something built into the API. You could give the feature request an upvote (#3147) but looks like that never got much community interest. I have built an extension to do this you could give a try in the meantime https://github.com/br41nslug/directus-extension-increment

thanks for the information. I checked  https://github.com/br41nslug/directus-extension-increment , please correct me if I am wrong,  the extension will modify the req data, by calculating the actual value, using the original data as input.

So basically, there are still two SQL queries, one for fetching the original data, another for setting the new data.

It's one API call, but still inside the API call, there are two SQL queries.

What I want to achieve is to use only ONE SQL query to update a column.

**Reply by br41nslug** [reply]
Yep that basic example still does 2 sql queries.

> What I want to achieve is to use only ONE SQL query to update a column.

You can very well do that with knex as you've pointed out but you'll have to build it :)
